### PR TITLE
[Macros] Implement peer declaration macros with expansion

### PR DESCRIPTION
**Pull request by DougGregor**
Peer declaration macros are applied to a declaration via an attribute,
but are responsible for generating "peer" declarations that go
alongside the declaration to which the attribute is attached.
Implement the `addCompletionHandler` example, which takes:

    @addCompletionHandler
    func f(a: Int, for b: String, _ value: Double) async -> String { }

and adds the following peer declaration:

    func f(a: Int, for b: String, _ value: Double, completionHandler: (String) -> Void) {
      Task {
        completionHandler(await f(a: a, for: b, value))
      }
    }

### CODE REVIEW

**Comment by DougGregor** [comment]
@swift-ci please test

File: Sources/_SwiftSyntaxMacros/ExpressionMacro.swift [hunk 1]
```diff
@@ -12,7 +12,6 @@
 
 import SwiftSyntax
 import SwiftParser
```
Comment (rxwei) [inline Sources/_SwiftSyntaxMacros/ExpressionMacro.swift:14]: Looks like SwiftParser isn't needed here either

Comment (DougGregor) [reply Sources/_SwiftSyntaxMacros/ExpressionMacro.swift:14]: You're right, the macro protocols have a bunch of extra imports. I've cleaned them up locally.

**Comment by DougGregor** [comment]
@swift-ci please test macOS

**Comment by DougGregor** [comment]
@swift-ci please test macOS

**Comment by DougGregor** [comment]
@swift-ci please test macOS

**Pull request merged**